manifests/update.pp

```diff
@@ -56,13 +56,20 @@
   } else {
     $_refresh = true
   }
+  $apt_update_successful = @("ASH"/$)
+    for try in $(seq ${apt::_update['tries']} -1 1); do
+      timeout ${apt::_update['timeout']} ${apt::provider} update && exit 0
+      [ \$try -eq 1 ] && exit \$?
+      sleep 1
+    done
+    - ASH
   exec { 'apt_update':
-    command     => "${apt::provider} update",
+    command     => "${apt::provider} update FAILED",
```

Comment:
Maybe this should run something more descriptive? When running `apt-get update FAILED` the output is `E: The update command takes no arguments`.

```suggestion
    command     => "/bin/false ${apt::provider} update FAILED",
```